### ISSUE TITLE
Fix false negatives on second run for cache and plugins

### ISSUE THREAD

**Issue by Yivan** [stylelint/stylelint#4715]
> Clearly describe the bug

Stylelint doesn't report warning of plugins on second run if cache activated

> What stylelint configuration is needed to reproduce the bug?

Install:
`npm install stylelint-no-unsupported-browser-features`

And add in .stylelintrc:
```
  "plugins": [
    "stylelint-no-unsupported-browser-features"
  ],
  "rules": {
    "plugin/no-unsupported-browser-features": [true, {
      "severity": "warning"
    }],
```

And active styelint `cache` to `true`

> Which version of stylelint are you using?

13.3.2

> How are you running stylelint: CLI, PostCSS plugin, Node.js API?

Node.js API

> Does the bug relate to non-standard syntax (e.g. SCSS, Less etc.)?

No

> What did you expect to happen?

On second run, see the same warning than in the first run

> What actually happened (e.g. what warnings or errors did you get)?

On the second run only warning on files having errors are reported

**Comment by jeddy3** [comment]
@Yivan Thanks for the report and for using the template.

I can reproduce this locally on the CLI with:

```json
{
  "plugins": ["stylelint-no-unsupported-browser-features"],
  "rules": {
    "plugin/no-unsupported-browser-features": [
      true,
      {
        "browsers": ["ie 6-8"],
        "severity": "warning"
      }
    ]
  }
}
```

```css
a {
  display: flex;
}
```

I've labelled the issue as ready to implement. Please consider [contributing](https://stylelint.io/contributing) if you have time.

**Comment by kimulaco** [comment]
I would like to work on this issue.

**Comment by jeddy3** [comment]
@kimulaco That'd be fantastic, thank you!

By the way, you're welcome to start working on any issue you like without commenting. As a Stylelint member, you've permission to change the `status: ready to implement` label to `status: wip`. That'll let other people know you're working on it.

**Comment by kimulaco** [comment]
@jeddy3 Thanks. I will update the label myself next time and start working!
